### ISSUE TITLE
Inserting named tensor into other fails

### ISSUE THREAD

**Issue by Quickblink** [pytorch/pytorch#49427]
## 🐛 Bug

Assigning a tensor to a view of another tensor (for insertion) fails because it implicitly calls tensor.view() which is not supported. This is so very basic functionality though so I wouldn't really consider this a feature request...

## To Reproduce

```
a = torch.zeros([1,2,3], names=('a', 'b', 'c'))
b = torch.zeros([2,3], names=('b', 'c'))

a[0] = b
```

produces

```
RuntimeError: view is not yet supported with named tensors. Please drop names via `tensor = tensor.rename(None)`, call the op with an unnamed tensor, and set names on the result of the operation.
```

## Expected behavior

I would expect this to work without name drop if the dimension names match accordingly.

## Environment

 - PyTorch Version (e.g., 1.0): 1.4.0

**Comment by leonvol** [comment]
I'll start working on this issue.

**Comment by leonvol** [comment]
Apparently, this issue seems to be fixed in the current 1.8.0a0 build. Versions before (also the current release 1.7.1) do not support that operation.

**Comment by zou3519** [comment]
I don't think someone specifically went in and fixed this in between 1.7 and 1.8. Therefore we should probably add a test for this behavior then, to prevent regressions. @leonvol would you be interested in doing that?

**Comment by leonvol** [comment]
Yes, I agree. I also did not see a specific PR addressing this. I am currently working on another fix, but if the issue remains open until I finished that one, I will take a look @zou3519.
